### PR TITLE
Use a database and JSON file for persistent, multi-day, live metrics.

### PR DESCRIPTION
**Pull request by cryptoquick**
This should be a superior approach that addresses a concern @crisdut raised about using a Lazy RwLock and also should lessen the burden on the server.

Supersedes #438

### CODE REVIEW

**Comment by josediegorobles** [comment]
Lots of tests are failing

**Comment by oacdutra** [comment]
> Lots of tests are failing

I will check this. The error being described is related to file manipulation.

File: src/carbonado/metrics.rs [hunk 1]
```diff
@@ -287,9 +315,102 @@ pub async fn update(path: &Path) -> Result<()> {
...
+    } else {
+        METRICS_DB
+            .open_tree(DB_DAYS)?
+            .insert(day.as_bytes(), &[1])?;
+        true
+    };
+
+    let dir = env::var("CARBONADO_DIR").unwrap_or("/tmp/bitmaskd/carbonado".to_owned());
+    let mut metrics: MetricsData =
+        serde_json::from_str(&fs::read_to_string(format!("{dir}/metrics.json")).await?)?;
```
Comment (oacdutra) [inline src/carbonado/metrics.rs:342]: When we run automated tests, no integration test initializes these files, which ends up causing the error.

Could we add a check here to see if the file exists?

**Review by oacdutra — CHANGES_REQUESTED** [review]

**Comment by cryptoquick** [comment]
@crisdut Can you check if bitmaskd isn't being instantiated in time for it to ensure the directory is initialized for backend integration tests?

**Comment by oacdutra** [comment]
> Can you check if bitmaskd isn't being instantiated in time for it to ensure the directory is initialized for backend integration tests?

We instance `bitmaskd` only for wasm32 tests. For integration tests, we call functions directly.

We need a verification like this:

```rust
    let dir = env::var("CARBONADO_DIR").unwrap_or("/tmp/bitmaskd/carbonado".to_owned());
    if !Path::new(&format!("{dir}/metrics.json")).exists()
        || !Path::new(&format!("{dir}/metrics.csv")).exists()
    {
        init(Path::new(&dir)).await?;
    }
```

**Review by oacdutra — APPROVED** [review]

**Review by josediegorobles — APPROVED** [review]
ACK

**Pull request merged**